src/main/java/seedu/address/model/person/Person.java

```diff
@@ -46,6 +46,23 @@ public Person(Name name, Phone phone, Email email, Address address,
         this.uniqueId = UUID.randomUUID(); // Generate a unique ID
     }
 
+    /**
+     * Constructor to create Person from memory using UUID
+     */
+    public Person(Name name, Phone phone, Email email, Address address,
+                  TelegramUsername telegramUsername, Set<Role> roles,
+                   UUID uniqueId) {
```

Comment:
Thanks for incorporating the UUID in the Person constructor

Reply:
In line with the other comment, I wi also create a static UUID lookup method so that the eventmanager is able to retrieve the saved persons.